src/objective-c/tests/UnitTests/GRPCCallOptionsTests.m

```diff
@@ -0,0 +1,161 @@
+  mutableOpt.additionalChannelArgs = @{@"extra_key" : @"extra_value"};
+  mutableOpt.PEMPrivateKey = kGRPCCallOptionsTestPEMPrivateKey;
+  mutableOpt.PEMRootCertificates = kGRPCCallOptionsTestPEMRootCertificates;
+  mutableOpt.PEMCertificateChain = kGRPCCallOptionsTestPEMCertificateChain;
+  mutableOpt.transportType = kGRPCCallOptionsTestTransportType;
+  mutableOpt.transport = kGRPCCallOptionsTestTransportID;
+  mutableOpt.hostNameOverride = kGRPCCallOptionsTestHostNameOverride;
+  mutableOpt.channelPoolDomain = kGRPCCallOptionsTestChannelPoolDomain;
+  mutableOpt.channelID = kGRPCCallOptionsTestChannelID;
+  return mutableOpt;
+}
+
+- (BOOL)isEqualToTestCallOptions:(GRPCCallOptions *)callOpt {
+  return [callOpt.serverAuthority isEqualToString:kGRPCCallOptionsTestServerAuthority] &&
+         callOpt.timeout == kGRPCCallOptionsTestTimeout &&
+         callOpt.flowControlEnabled == kGRPCCallOptionsTestFlowControl &&
+         [callOpt.oauth2AccessToken isEqualToString:kGRPCCallOptionsTestOAuth2Token] &&
+         [callOpt.initialMetadata isEqualToDictionary:@{@"key" : @"value"}] &&
+         [callOpt.userAgentPrefix isEqualToString:kGRPCCallOptionsTestUserAgentPrefix] &&
+         [callOpt.userAgentSuffix isEqualToString:kGRPCCallOptionsTestUserAgentSuffix] &&
+         callOpt.responseSizeLimit == kGRPCCallOptionsTestResponseSizeLimit &&
+         callOpt.compressionAlgorithm == kGRPCCallOptionsTestCompressionAlgorithm &&
+         callOpt.retryEnabled == kGRPCCallOptionsTestRetryEnabled &&
+         callOpt.maxRetryInterval == kGRPCCallOptionsTestMaxRetryInterval &&
+         callOpt.minRetryInterval == kGRPCCallOptionsTestMinRetryInterval &&
+         callOpt.retryFactor == kGRPCCallOptionsTestRetryFactor &&
+         callOpt.keepaliveTimeout == kGRPCCallOptionsTestKeepAliveTimeout &&
+         callOpt.keepaliveInterval == kGRPCCallOptionsTestKeepAliveInterval &&
+         callOpt.connectMaxBackoff == kGRPCCallOptionsTestConnectMaxBackoff &&
+         callOpt.connectMinTimeout == kGRPCCallOptionsTestConnectMinTimeout &&
+         callOpt.connectInitialBackoff == kGRPCCallOptionsTestConnectInitialBackoff &&
+         [callOpt.additionalChannelArgs isEqualToDictionary:@{@"extra_key" : @"extra_value"}] &&
+         [callOpt.PEMPrivateKey isEqualToString:kGRPCCallOptionsTestPEMPrivateKey] &&
+         [callOpt.PEMCertificateChain isEqualToString:kGRPCCallOptionsTestPEMCertificateChain] &&
+         [callOpt.PEMRootCertificates isEqualToString:kGRPCCallOptionsTestPEMRootCertificates] &&
+         callOpt.transportType == kGRPCCallOptionsTestTransportType &&
+         callOpt.transport == kGRPCCallOptionsTestTransportID &&
+         [callOpt.hostNameOverride isEqualToString:kGRPCCallOptionsTestHostNameOverride] &&
+         [callOpt.channelPoolDomain isEqualToString:kGRPCCallOptionsTestChannelPoolDomain] &&
+         callOpt.channelID == kGRPCCallOptionsTestChannelID;
+}
```

Comment:
Is there a reason not to implement -[GRPCCallOptions isEqual:]? That might be useful to implement and could replace this helper method.

Reply:
Thanks Stephane, yes we have a TODO to add isEqual & hash support for the call option objects. A few of the properties on the objects are non-hashable at the moment (e.g. tokenProvider) and we will be resolving those first.